### ISSUE TITLE
Intuit's Oauth2 support- Request 3- Keys for Authorize url have _ 

### ISSUE THREAD

**Issue by nimisha84** [DuendeArchive/IdentityModel#147]
Intuit's spec/implementation https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/openid-connect#prerequisites
uses _ each of the keys used in AuthorizeUrl call-
GET https://appcenter.intuit.com/connect/oauth2?
client_id=Q3ylJatCvnkYqVKLmkH1zWlNzNWB5CkYB36b5mws7HkKUEv9aI&
response_type=code&
scope=openid%20email&
redirect_uri=https://www.mydemoapp.com/oauth-redirect&
state=security_token%3D138r5719ru3e1%26url%3Dhttps://www.mydemoapp.com/oauth-redirect&

Using object extra is the only way of supporting this variation in names?  Also, clientId and responseType is required in IdentityModel, so can't leave those out while building the Authorize request url.
Our devs would prefer standard keys already defined in the class. Please suggest.

**Comment by leastprivilege** [comment]
I just checked - in `RequestUrlExtensions.CreateAuthorizeUrl` you should have everything you need.

If you want something more domain specific I would recommend simply building your own extension method over `RequestUrl`.

**Comment by nimisha84** [comment]
Yes, I will build the extension wherever I can after using the binaries( but this might take some time as I will be on parental leave for about 1.5 months. Might do some minor updates to Oauth2 SDK but rewrite I will be able to start only after my leaves.)  Thanks!
Wanted to know which branch is the change to RequestUrlExtensions.CreateAuthorizeUrl checked in?

**Comment by nimisha84** [comment]
The same issue of _ is there for Intuit's Get token call-.
IdentityModel binaries have redirectUri as required whereas Intuit has redirect_uri. Created new issue to track this.

**Comment by leastprivilege** [comment]
well - identitymodel is spec compliant. These are the official names.

**Comment by nimisha84** [comment]
You are right, but unfortunately I cannot use them due to the underlying implementation. What do you suggest then? Just like above, will be supporting CreateAuthorizeUrl or RequestTokenAsync with 'all' not required params also so that if I send only extra object values, only those are used?

**Comment by leastprivilege** [comment]
yea - extension method for RequestUrl and use RequestTokenAsync - then pass in all your custom values. give that a try. Please open a new issue if there is a problem.

**Comment by nimisha84** [comment]
Thanks!
